csh-pings-backend/src/main/java/edu/rit/csh/pings/external/DiscordService.java

```diff
@@ -0,0 +1,121 @@
+    private final Collection<GatewayIntent> intents = EnumSet.of(GatewayIntent.GUILD_MEMBERS, GatewayIntent.GUILD_MESSAGES);
+    private JDA jda = JDABuilder.create(DISCORD_TOKEN, intents).build();
+
+    @PostConstruct
+    public void readTemplates() {
+        try {
+            this.pingTemplate = readFully(new ClassPathResource("discord_ping_template.txt").getInputStream());
+            this.verificationTemplate = readFully(new ClassPathResource("discord_verification_template.txt").getInputStream());
+        } catch (IOException e) {
+            throw new IOError(e);
+        }
+    }
+
+    @PostConstruct
+    public void setup() {
+        try {
+            jda.awaitReady();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        // get the guilds the bot is in
+        jda.getGuilds().forEach(guild -> {
+            // get the users in the guild
+            guild.getMembers().forEach(member -> {
+                // add the user to the set
+                usernameToId.put(member.getUser().getName() + "#" + member.getUser().getDiscriminator(), member.getUser());
+            });
+        });
+        sendMessageTest("Creeper", "3621", "Discord bot is online!");
+    }
+
+    @PreDestroy
+    public void shutdown() throws InterruptedException {
+        sendMessageTest("Creeper", "3621", "Discord bot is shutting down!");
+        jda.shutdown();
+        jda.awaitStatus(JDA.Status.SHUTDOWN);
+        log.info("Discord bot has shut down successfully");
+    }
+
+    public void sendPing(Route route, DiscordServiceConfiguration config, String body) {
+        this.log.info("Sending Ping to D-@" + config.getDiscordUsername() + "#" + config.getDiscordDiscriminator() + ", CSH-@" + config.getUsername());
+        final Application app = route.getApplication();
+        final String send = this.pingTemplate
+                .replace("%%%BODY%%%", body)
+                .replace("%%%APPLICATION%%%", app.getName());
+        sendMessage(config, send);
+    }
+
+    @Override
+    public void sendVerification(DiscordServiceConfiguration config) {
+        final VerificationRequest vr = this.verificationRequestManager.generateVerification(config);
+        this.log.debug("Created VR");
+        final String url = this.url + (this.url.endsWith("/") ? "" : "/") + "verify?token=" + vr.getToken();
+        final String verificationText = this.verificationTemplate.replace("%%%LINK%%%", url).replace("%%%USER%%%", config.getUsername());
+        // send the verification message to the user
+        sendMessage(config, verificationText);
+    }
+
+    public void sendMessage(DiscordServiceConfiguration config, String message) {
+        User user = usernameToId.get(config.getDiscordUsername() + "#" + config.getDiscordDiscriminator());
+        if (user == null) {
+            log.error("Could not find user " + config.getDiscordUsername() + "#" + config.getDiscordDiscriminator());
+            return;
```

Comment:
If the user does not exist, please throw an `IllegalArgumentException`, because this method will leave an Orphan DB entry for the configuration even if it isn't invalid.